src/Samples/SarifToCsv/Program.cs

```diff
@@ -150,6 +149,13 @@ public static void ConvertSarifLog(JsonSerializer serializer, string sarifFilePa
             writers["Location.Region.StartLine"] = (writer, result, pLoc) => { writer.Write(pLoc.Region?.StartLine ?? -1); };
             writers["Location.Region.Tags"] = (writer, result, pLoc) => { writer.Write(String.Join("; ", ((IEnumerable<string>)pLoc.Region?.Tags) ?? Array.Empty<string>())); };
 
+            writers["Properties.MatchState"] = (writer, result, pLoc) => 
+            {
+                string state;
+                if(!result.TryGetProperty("matchState", out state)) { state = null; }
```

Comment:
>{ state = null; } [](start = 68, length = 17)

this assignment isn't necessary. state will be set to null if TryGetProperty fails. you can't return from a method with an out parameter without initializing it.

Reply:
True. I like to be explicit in cases like this so one doesn't have to look up what the method returns (though out is a logical default). It also looks a little weird when one calls "TryXX" and then ignores the return value.